OpenPanelSDK/src/main/java/com/dev/openpanelsdk/OpenPanel.kt

```diff
@@ -238,23 +204,177 @@ open class OpenPanel(protected val options: Options) {
         currentQueue.forEach { send(it) }
     }
 
-    open fun setupAutomaticTracking() {
-        // Base implementation (empty)
+    private fun getDefaultEventProperties(): Map<String, Any> {
+        val ret = mutableMapOf<String, Any>()
+
+        ret["op_lib"] = "android"
+        ret["lib_version"] = sdkVersion
+
+        // For querying together with data from other libraries
+        ret["os"] = "Android"
+        ret["os_version"] = Build.VERSION.RELEASE ?: "UNKNOWN"
+
+        ret["manufacturer"] = Build.MANUFACTURER ?: "UNKNOWN"
+        ret["brand"] = Build.BRAND ?: "UNKNOWN"
+        ret["model"] = Build.MODEL ?: "UNKNOWN"
+
+        val displayMetrics = mSystemInformation?.displayMetrics
+        ret["screen_dpi"] = displayMetrics?.densityDpi ?: "UNKNOWN"
+        ret["screen_height"] = displayMetrics?.heightPixels ?: "UNKNOWN"
+        ret["screen_width"] = displayMetrics?.widthPixels ?: "UNKNOWN"
+
+        val applicationVersionName = mSystemInformation?.appVersionName
+        if (applicationVersionName != null) {
+            ret["app_version"] = applicationVersionName
+        }
+
+        val applicationVersionCode = mSystemInformation?.appVersionCode
+        if (applicationVersionCode != null) {
+            val applicationVersion = applicationVersionCode.toString()
+            ret["app_release"] = applicationVersion
+            ret["app_build_number"] = applicationVersion
```

Comment:
Can you use `__buildNumber`?

For version could you use `__version`?

I have these for react-native and would be nice to align